generate-geodatabase-replica-from-feature-service/src/main/java/com/esri/arcgismaps/sample/generategeodatabasereplicafromfeatureservice/MainActivity.kt

```diff
@@ -0,0 +1,256 @@
+                mapView.visibleArea?.let { polygon ->
+                    // start the geodatabase generation process
+                    generateGeodatabase(geodatabaseSyncTask, map, polygon.extent)
+                }
+            }
+        }
+    }
+
+    /**
+     * Starts a GeodatabaseSyncTask, runs a GenerateGeodatabaseJob and saves
+     * the geodatabase file into local storage
+     *
+     * @param geodatabaseSyncTask the GeodatabaseSyncTask object
+     * @param map mapView to render the feature layers on
+     * @param extents the selected extents of the mapView
+     */
+    private fun generateGeodatabase(geodatabaseSyncTask: GeodatabaseSyncTask, map: ArcGISMap, extents: Envelope) {
+        // clear any layers already on the map
+        map.operationalLayers.clear()
+        // clear all symbols drawn
+        graphicsOverlay.graphics.clear()
+
+        // create a boundary representing the extents selected
+        val boundary = Graphic(extents, boundarySymbol)
+        // add this boundary to the graphics overlay
+        graphicsOverlay.graphics.add(boundary)
+
+        lifecycleScope.launch {
+            // create generate geodatabase parameters for the selected extents
+            val defaultParameters = geodatabaseSyncTask.createDefaultGenerateGeodatabaseParameters(extents).getOrElse {
+                // show the error and return if the task fails
+                showError("Error creating geodatabase parameters")
+                return@launch
+            }
+
+            // set return attachments option to false
+            defaultParameters.returnAttachments = false
+            // create a generate geodatabase job
+            geodatabaseSyncTask.createGenerateGeodatabaseJob(defaultParameters, geodatabaseFilePath).run {
+                // create a dialog to show the jobs progress
+                val dialog = createProgressDialog(this).apply {
+                    // show the dialog
+                    show()
+                    // allow it to be cancellable
+                    setCancelable(true)
+                    // disable dismissal when tapped outside the dialog
+                    setCanceledOnTouchOutside(false)
+                }
+                // launch a progress collector to display progress
+                launch {
+                    progress.collect { value ->
+                        // update the progress bar and progress text
+                        progressDialog.progressBar.progress = value
+                        progressDialog.progressTextView.text = "$value%"
+                    }
+                }
+                // start the generateGeodatabase job
+                start()
+                // if the job completed successfully, get the geodatabase from the result
+                val geodatabase = result().getOrElse {
+                    // show an error and return if job failed
+                    showError("Error fetching geodatabase")
```

Comment:
```suggestion
                    showError("Error fetching geodatabase: ${it.message}")
```
Could add the throwable message to the error